src/mono/wasm/runtime/driver.c

```diff
@@ -823,6 +826,10 @@ void mono_wasm_ensure_classes_resolved ()
 		safehandle_class = mono_class_from_name (mono_get_corlib(), "System.Runtime.InteropServices", "SafeHandle");
 		resolved_safehandle_class = 1;
 	}
+	if (!voidtaskresult_class && !resolved_voidtaskresult_class) {
+		voidtaskresult_class = mono_class_from_name (mono_get_corlib(), "System.Threading.Tasks", "VoidTaskResult");
```

Comment:
This is fragile.  There's no way to support any `Task<T>` regardless of its type parameter?

Reply:
VoidTaskResult is a struct, we don't handle structs. (I'm working on that)

Reply:
> I'm working on that

Ok

Reply:
Yeah, the less fragile fix in in process. I did the simple fix after digging into it so we can unblock/write some tests.